src/wccp2.cc

```diff
@@ -629,9 +627,8 @@ wccp2_check_security(struct wccp2_service_list_t *srv, char *security, char *pac
     /* If execution makes it here then we have an MD5 security */
 
     /* The password field, for the MD5 hash, needs to be 8 bytes and NUL padded. */
-    memset(pwd, 0, sizeof(pwd));
-    strncpy(pwd, srv->wccp_password, sizeof(pwd));
-    pwd[sizeof(pwd) - 1] = '\0';
+    memset(pwd, 0, WCCP2_PASSWORD_LEN);
+    xstrncpy(pwd, srv->wccp_password, WCCP2_PASSWORD_LEN);
```

Comment:
Again here, please leave the sizeof(pwd) as the size parameter.

Reply:
Done.